mentat/parsers/block_parser.py

```diff
@@ -197,3 +197,71 @@ def _add_code_block(
             )
         )
         return ""
+
+    def file_edits_to_llm_message(self, generated: tuple[str, list[FileEdit]]) -> str:
+        """
+        Inverse of stream_and_parse_llm_response;
+        takes in the generated message and file edits and returns the original message
+        """
+        ans = generated[0].split("@")[0]  # BlockParserIndicator.Start.value)[0]
```

Comment:
Hold on, I'm a bit confused on what's going on here; is generated[0] just the actual message that the model output (that we're trying to get back, but potentially in a different format)?  If so, the @ won't always be the end of the 'conversation' that gets generated (since some formats have different markers). We could do one of 2 things: change parsers to output the actual 'conversation' that the model output (this would also catch conversation in between edits, which you drop right now), or we could check for any format's start marker.

Reply:
I like that idea a lot! Definitely is a bit messy right now haha